src/main/java/sqlline/Application.java

```diff
@@ -200,6 +201,7 @@ public String getVersion() {
    *
    * @see #DEFAULT_DRIVERS
    */
+  @Deprecated
```

Comment:
@snuyanzin how list of known drivers can be overridden without using depicted method?

Reply:
Could you please clarify what you mean?

Before this PR known drivers used to filter classes from classpath and to load drivers if they are matched to known drivers.
After PR changes java service provider functionality [1] will be used, i.e. java could go through jars looking for META-INF/services/java.sql.Driver and if success then load the driver.
So each driver which has such file inside its jar becomes known. It will allow to add more drivers in classpath and load them automatically without code changes

[1] https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html

Reply:
For Drill we were overriding all known drivers to only one - "org.apache.drill.jdbc.Driver" since Drill SqlLine is intended only for Apache Drill.

Reply:
Looks like for your case it is required to have limit which could be done via _Application#initDrivers_. What about logic like if _Application#initDrivers_ is empty then any driver found via servicelocator is allowed. If _Application#initDrivers_ is not empty then only drivers which are present in both _Application#initDrivers_ and servicelocator are allowed.

However the name _Application#initDrivers_ becomes not matching to what it started to do. So it makes sense to add a new method with more suitable name and leave it deprecated and made call to the new one as it is public.

Reply:
Sounds good, just indicate in Javadoc which new method should be used.